site/app/controllers/DockerInterfaceController.php

```diff
@@ -75,4 +80,122 @@ public function showDockerInterface(): MultiResponse {
             )
         );
     }
+    /**
+     * @Route("/admin/add_image", methods={"POST"})
+     * @return JsonResponse
+     */
+    public function addImage(): JsonResponse {
+        $user = $this->core->getUser();
+        if (is_null($user) || !$user->accessFaculty()) {
+            return new MultiResponse(
+                JsonResponse::getFailResponse("You don't have access to this endpoint."),
+                new WebResponse("Error", "errorPage", "You don't have access to this page.")
+            );
+        }
+        
+        if (!isset($_POST['image'])) {
+            $this->core->addErrorMessage("Image not set");
+            return JsonReponse::getErrorResponse("Image not set");
+        }
+        if (!isset($_POST['capability'])) {
+            $this->core->addErrorMessage("Capability not set");
+            return JsonReponse::getErrorResponse("Capability not set");
+        }
+
+        // check for proper format
+        $match = preg_match('/^[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+\/[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+:[a-zA-Z0-9][a-zA-Z0-9._-]{0,127}$/', $_POST['image']);
+        if ($match == 0) {
+            $this->core->addErrorMessage("Improper docker image name");
+            return JsonReponse::getErrorResponse("Improper docker image name");
+        }
+        $image_arr = explode(":", $_POST['image']);
+        // ping the dockerhub API to check if docker exists
+        $url = "https://registry.hub.docker.com/v1/repositories/".$image_arr[0]."/tags";
```

Comment:
When I tried to use file_get_contents, the function call returns false. Do you know why this might be?

Reply:
It would depend on what type of request the docker API endpoint is expecting, so it might not work if it's not a plain get request when a resource pointing to the data you want like for example it is expecting json or wants a specific header set

That said I can't find more about the API endpoint since it looks like v1 is deprecated? If this lookup is being added now should the v2 API be used instead

Reply:
The problem with the v2 API is that it requires authentication to be used, after talking about this with Barb, we decided that we will use v1 for now.